app/models/result.rb

```diff
@@ -0,0 +1,9 @@
+# Immutable result object for use in services
+class Result < Data.define(:value, :error, :success)
+  # Set default attribute values
+  def initialize(value: nil, error: nil, success: false)
+    super
+  end
+
+  alias_method :success?, :success
```

Comment:
Do we need a `success` attribute? Why wouldn't we just define a method?

```ruby
def success? = error.nil?
```

Reply:
Yea I was debating this internally and wasn't sure which was better. WIll fix.